City-of-Helsinki/berth-reservations-admin
# VEN-198 | Table's global filter

## Description :sparkles:
- Use a flexible layout for the Table
- Support global filter
- Adjust the Table's styles
- Add a filter to the Harbors page
- Define the columns' widths ratios of the main tables
- Added micro animation to the sorting arrow
- Added scroll bars to cells when the data overflow

## Issues :bug:

### Closes :no_good_woman:
- [VEN-198](https://helsinkisolutionoffice.atlassian.net/browse/VEN-298) Table search bar for the Harbours list table

### Related :handshake:

## Testing :alembic:

### Automated tests :gear:️
https://github.com/City-of-Helsinki/berth-reservations-admin/blob/a8c026fcca6ed0f1a46ed7a2040e06b2e96df10b/src/domain/harbors/harborsTableTools/HarborsTableTools.test.jsx#L2

### Manual testing :construction_worker_man:
- Go the harbors page
- Type down some value in the text field above the table
- The table's data should be filtered by the search value
- When there are no matches, the table should display a row that states that

Note: it is a global filter, so you are not restricted by one column

## Screenshots :camera_flash:
![Screen-Recording-2020-04-06-at-1](https://user-images.githubusercontent.com/23040926/78570819-256dc100-782e-11ea-9da3-06a4744feb30.gif)

## Additional notes :spiral_notepad:
